Feature/reusable modal

JaneMoroz opened the pull request:
# Description

Created a reusable modal with children props.
Added a temporary buttons component to the root page to trigger the modals. 
Modals can be closed by:
- clicking a close button at corner
- pressing the ESC key
- clicking outside of the modal

## Issue link
[Reusable Modal](https://trello.com/c/iJz9HeNq/4-reusable-modal)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration


# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Dan-Y-Ko commented:
Please see the branch feature/reusable-modal-refactor for suggested refactor for this feature.

JaneMoroz commented:
@Dan-Y-Ko 
- fixed daisyUI modal logic, now all modals don't render on load
- used children instead of bodyContent prop but still decided to not to move modalBody into separate component, not sure if it can be server component
- moved redux stuff to the base modal. But well... not sure about it though, because at first I assumed that each modal will still use dispatch or router, maybe we'll have to use `isLoading` from the store... so thought no need to use redux stuff twice, but that just assumptions, don't know how we gonna handle forms yet

JaneMoroz commented:
Added a server modal, open/close functionality is triggered by search params.
